Adding ADR for describing the code structuring discussion

carolyncole opened the pull request:
fixes #135

escowles commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+# Structure of Classes Beyond Standard RailsMVC
+
+There are many ways to structure classes beyound the standard rails Model View Controller (MVC) structure, therefore to help make the project maintainable in the future we have come up with the following standards.
```
Typo: `beyound`

escowles commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+* Use a Decorators directory `app/decorators`
+  * Decorators should be used for read only operations that modify the data for viewing purposes.
+
+* Use a ChangeSet directory `app/change_set`
```
Should that be `change_sets`?

escowles commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+
+* Use a ChangeSet directory `app/change_set`
+  * ChangeSet should be used for reformatting the data for purposes of showing it to user/system for modification.
+  * We have kept the chnage set concept since these classes can also be useful for data import not just user facing formas.
```
Typo: `chnage`

hackmastera replied:
also "formas"

escowles commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+We currently believe that the buisness logic operations that are required for the system will be small enough to live directly on their primary model.  If in the future the models become to bloated we reccomend creating a BusinessRules directory `app/business_rules`.
+
+Currrently we plan to put the rules here:
+* Other attendees could go on the RecurringEvent #102
```
Should this be a link to an issue?

escowles commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+# Structure of Classes Beyond Standard RailsMVC
```
Do you want to have the header/format of the other projects that have ADRs?  e.g., https://github.com/pulibrary/figgy/blob/master/architecture-decisions/0001-document-architecture-decisions.md

carolyncole replied:
Sure!  @escowles I'm a bit lost on the consequences section.  What would you expect to see there for a document like this?

escowles replied:
I've sometimes struggled with the consequences section in other ADRs.  For this code organization, I think it's probably something along the lines of:
* This organization makes sense to us, but it's a little different from Rails standard, and a little more like Figgy
* Devs who are familiar with these patterns will understand the division and patterns being used
* Devs who aren't familiar with `change_sets` or `services` might take more time to find things

hackmastera commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+  * Notifications will be put in the mailer directory so that we can stick with the standard defined by ActionMailer
+
+## Possible Future needs
+We currently believe that the buisness logic operations that are required for the system will be small enough to live directly on their primary model.  If in the future the models become to bloated we reccomend creating a BusinessRules directory `app/business_rules`.
```
typo: "to" should be "too"

hackmastera replied:
Add to this section something like "`business_rules` would be a corollary to `services`, for complex or model-spanning read-only logic".

hackmastera commented on adr/structure_classes.md:
```diff
@@ -0,0 +1,28 @@
+## Possible Future needs
+We currently believe that the buisness logic operations that are required for the system will be small enough to live directly on their primary model.  If in the future the models become to bloated we reccomend creating a BusinessRules directory `app/business_rules`.
+
+Currrently we plan to put the rules here:
```
I would remove this section since it's very specific to tickets and upcoming implementations at this point in time, whereas I think the ADR should be more timeless.

carolyncole commented:
@hackmastera & @escowles You will be happy to know I installed a spell checker into my editor.  It caught a number of mistakes including the ones you noted (although not the grammar ones).  I Added the headers and updated the content.  I believe it reflects all the requests.

hackmastera commented on adr/001_structure_classes.md:
```diff
@@ -0,0 +1,39 @@
+## Decision
+
+* Avoid Helpers
+  * The rational behind this decision is standard rails helpers are hard to test efficiently and are global so they can have unintended consequences.
```
rationale

hackmastera commented on adr/001_structure_classes.md:
```diff
@@ -0,0 +1,39 @@
+## Consequences
+
+* This organization makes sense to us, but it's a little different from Rails standard, and a little more like Figgy
+* Developers who are familiar with these patterns will understand the division and patterns being used
```
missing periods on a couple of sentences

carolyncole commented:
@hackmastera made the last couple updates.  Thanks for copy editing!
